shuup/utils/money.py

```diff
@@ -60,3 +63,30 @@ def unit_matches_with(self, other):
 
     def new(self, value):
         return type(self)(value, currency=self.currency)
+
+    def as_rounded(self, digits=None):
+        """
+        Returns the value rounded to the given `digits` if specified,
+        or to the fetched `decimal_places` from the Currency
+        loaded from db using the `currency` attribute of this instance.
+
+        In case of non-existent currency in the db,
+        the fallback `digits` will always be 0.
+
+        :param digits int:
+          The number of digits to round
+
+        :return A new value rounded to the given digits
+        """
+
+        if digits is None:
+            # the digits info is not cached yet, fetch it
+            if self.currency not in digits_cache:
+                from shuup.core.models._currencies import Currency
+
+                currency = Currency.objects.filter(code=self.currency).first()
+                digits_cache[self.currency] = currency.decimal_places if currency else 0
+
+            return self.new(numbers.bankers_round(self.value, digits_cache[self.currency]))
+        else:
+            return self.new(numbers.bankers_round(self.value, digits))
```

Comment:
* Don't add shuup.core.models.Currency stuff to Money class. See below for better way to do this.
* The method should have `rounding` param defaulting to ROUND_HALF_EVEN. (No need to use `bankers_round` btw., just use `Decimal.quantize`)
* Default precision should be 2 digits rather than 0, because it's the most popular money amount precision around the world.

Rather than adding shuup.core stuff to Money, just add accessors for setting the precisions, e.g.
```python
class Money(...):
    default_precision = decimal.Decimal('0.01')
    precisions = digits_to_precisions({
        'USD': 2, 'EUR': 2,  # sensible defaults, add more of these from some nice source like wikipedia...
    })
    @classmethod
    def set_currency_precisions(cls, digits):
        """
        Set default precisions for rounding money amounts.

        :param digits: Map from currency code to number of digits
        :type digits: dict[str,int]
        """
        assert isinstance(digits, dict)
        assert all(isinstance(x, int) for x in digits.values())
        cls.precisions = digits_to_precisions(digits)

    def as_rounded(self, digits=None, rounding=ROUND_HALF_EVEN):
        precision = self.precisions.get(self.currency, self.default_precision)
        return self.new(self.value.quantize(precision, rounding=rounding))


def digits_to_precisions(mapping):
    return {
        key: decimal.Decimal('0.1') ** digits
        for (key, digits) in mapping
    }
```

Then make sure that the precisions are set when shuup.core is loaded, e.g. by setting them in